src/main/java/org/opensearch/security/resources/ResourcePluginInfo.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.resources;
+
+// CS-SUPPRESS-SINGLE: RegexpSingleline get Resource Sharing Extensions
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import org.opensearch.security.spi.resources.ResourceProvider;
+import org.opensearch.security.spi.resources.ResourceSharingExtension;
+
+/**
+ * This class provides information about resource plugins and their associated resource providers and indices.
+ * It follows the Singleton pattern to ensure that only one instance of the class exists.
+ *
+ * @opensearch.experimental
+ */
+public class ResourcePluginInfo {
+
+    private final Map<String, ResourceProvider> resourceProviderMap = new HashMap<>();
+    private final Set<String> resourceIndices = new HashSet<>();
+
+    private final Set<ResourceSharingExtension> resourceSharingExtensions = new HashSet<>();
+
+    public void setResourceProviders(Map<String, ResourceProvider> providerMap) {
+        resourceProviderMap.clear();
+        resourceProviderMap.putAll(providerMap);
+    }
+
+    public void setResourceIndices(Set<String> indices) {
+        resourceIndices.clear();
+        resourceIndices.addAll(indices);
+    }
+
+    public void setResourceSharingExtensions(Set<ResourceSharingExtension> extensions) {
+        resourceSharingExtensions.clear();
```

Comment:
Why clear()? Can this be designed in a way that it is set once.

Reply:
it is used only in 1 place. I have removed this redundant clear.